refactor(slice-machine-ui): use `Table` from `@prismicio/editor-ui`

angeloashmore opened the pull request:
## Context

DT-2019

## The Solution

- Replace use of `Table` with `@prismicio/editor-ui`'s `Table`.
- Remove the empty space around the "Repeatable" and "Single" document icons.
- Remove the `Table` component set.
- Remove the `CustomTypesTablePage` Vitest tests. The tests broke as a result of this PR, and we already have Playwright tests to cover the page. It isn't worth our time to fix the Vitest version.

## Impact / Dependencies

N/A

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

## Preview

<img width="989" alt="image" src="https://github.com/prismicio/slice-machine/assets/8601064/f6e54fb8-02a4-448f-abb0-0e36abcf7da4">

angeloashmore committed:
refactor(slice-machine-ui): use `Table` from `@prismicio/editor-ui`

angeloashmore committed:
test: update test locator for new table component

angeloashmore committed:
test: remove `CustomTypesTablePage` tests since Playwright already covers them

bapmrl commented on packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTable.module.css:
```diff
@@ -0,0 +1,4 @@
+.ellipsis {
```
:warning: You can replace `ellipsis` with the `Text` component's `noWrap` prop.

angeloashmore replied:
Oh perfect! I didn't realize we had that built in.

bapmrl commented on packages/slice-machine/src/icons/ReusableIcon.tsx:
```diff
@@ -1,23 +1,17 @@
-    />
+export const ReusableIcon: FC<SVGProps<SVGSVGElement>> = (props) => (
+  <svg viewBox="0 0 20 24" xmlns="http://www.w3.org/2000/svg" {...props}>
+    <g fill="none">
```
:question: Have you done something different to export the SVG? I'm asking because the elements have changed a bit. For example, the `g` element wasn't used before.

angeloashmore replied:
Good catch. I re-exported the SVGs without the extra embedded whitespace. The whitespace was causing sizing issues with `Table` since the component requires a set column template.

I ran the re-exported SVG through SVGO, and I guess it figured the `<g>` was necessary. I removed them manually and the icons seem to work the same as with them.

bapmrl commented on packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTable.tsx:
```diff
@@ -82,49 +87,80 @@ export const CustomTypesTable: FC<CustomTypesTableProps> = ({
-      <TableBody>
-        {sortedCustomTypes.map((customType: CustomType) => {
-          const { repeatable, label, id } = customType;
+    <div>
```
:bulb: I'm not asking you to update the `@prismicio/editor-ui` package, but it's unfortunate the `Table` component has `flex-grow: 1` on its `.root` class. It makes `Table` difficult to use in contexts where it shouldn't take the full height of the container. CC @AlexGalays.

angeloashmore replied:
I was surprised by that as well. There must be a reason why it has that by default, but it would be great to remove it. For now, the extra `<div>` prevents `Table` from growing.

xrutayisire replied:
Arf, not super great true, did you asked why to editor @angeloashmore?

Anyway, should not block a merge we can quickly remove it the day they remove it also

bapmrl commented on packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTable.tsx:
```diff
@@ -82,49 +87,80 @@ export const CustomTypesTable: FC<CustomTypesTableProps> = ({
+              >
+                <TableCell>
+                  {repeatable ? (
+                    <ReusableIcon width={20} />
```
:bulb: You could use the matching theme token here.

angeloashmore replied:
Done!

bapmrl reviewed (changes requested):
Excellent!

🕵️‍♂️ I've only noticed that the `Button`'s shadow is clipped, but I don't understand why. I don't think this is a blocking issue though.

<img width="1047" alt="Screenshot 2024-05-06 at 11 02 10" src="https://github.com/prismicio/slice-machine/assets/8003503/c5c7eb3a-7935-4e16-a448-f82ca0ae9ba1">

angeloashmore committed:
refactor: replace `ellipsis` with `noWrap`

angeloashmore committed:
refactor: remove `<g>` from SVGs

angeloashmore committed:
refactor: use `theme.space[20]` over a literal value

filip-prismic commented:
> Excellent!
> 
> 🕵️‍♂️ I've only noticed that the `Button`'s shadow is clipped, but I don't understand why. I don't think this is a blocking issue though.
> 
> <img alt="Screenshot 2024-05-06 at 11 02 10" width="1047" src="https://private-user-images.githubusercontent.com/8003503/328131508-c5c7eb3a-7935-4e16-a448-f82ca0ae9ba1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUzMjQzNjQsIm5iZiI6MTcxNTMyNDA2NCwicGF0aCI6Ii84MDAzNTAzLzMyODEzMTUwOC1jNWM3ZWIzYS03OTM1LTRlMTYtYTQ0OC1mODJjYTBhZTliYTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTBUMDY1NDI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjMwZjUzNDFlOTZmYzU0NmFkYWM2N2Q0MDFhZjQ3ODVkNDZiZjU3YTliNDlhMGQ2YTgxYzNjMWUwYjk2MmYxOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dqgabx5BnJTH6z83HGbHqgy6YqLg8G2j9EaybrcxxiQ">

It is because TableCell has overflow class assigned by default and the outline spans beyond element bounds. Maybe we could provide prop to remove it, bcs in the cell like this you don't need it.
```css
.overflow {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}
```

angeloashmore committed:
fix: Prevent focus indicator clipping

angeloashmore commented:
> It is because TableCell has overflow class assigned by default and the outline spans beyond element bounds. Maybe we could provide prop to remove it, bcs in the cell like this you don't need it.
> 
> ```css
> .overflow {
>   text-overflow: ellipsis;
>   overflow: hidden;
>   white-space: nowrap;
> }
> ```

Exposing a prop that controls the overflow feels like a leaky abstraction. Throwing a different idea out: what if we removed the 16px inline padding and moved it to the first and last cells in a row? That would prevent the overflow clipping.

Unfortunately, it means the first and last column sizes need to increase by 16px, which may have negative consequences:

```diff
- <Table columnLayout="28px 1fr 1fr 1fr 40px">
+ <Table columnLayout="44px 1fr 1fr 1fr 56px">
```

angeloashmore commented:
@bapmrl I fixed the issue by increasing the table column by 2px: https://github.com/prismicio/slice-machine/pull/1362/commits/2dc6d5a48fb2c3f8d62697e16d637d6ebaf82f8c

I think it's an okay solution for now. The change is barely visible:

<img width="729" alt="image" src="https://github.com/prismicio/slice-machine/assets/8601064/0f4a0b2f-8adc-41c5-a89d-531770f3691c">

xrutayisire reviewed (approved):
LGTM! Let's merge and keep open the two small things that are not sure good because. of current editor Table implementation.

filip-prismic commented:
> what if we removed the 16px inline padding and moved it to the first and last cells in a row? That would prevent the overflow clipping.

@angeloashmore  seems like patch for a problem we should not be dealing with. I would rather ask why the overflow is there for all the columns  by default in the first place. Seems weird to me :D 

But agree this one shouldn't be a blocker for you and 2px size change is a good enough right now.

angeloashmore commented:
@filip-prismic I quickly looked at it closer and I agree; it doesn't seem like the overflow styles are needed. It could be something handled by the cell's children instead.

But let's move forward with the 2px fix for now.
